stl/team_aster/s1mplebug/src/task_1/list.cc

```diff
@@ -0,0 +1,234 @@
+#if 0
+// node struct with two pointers.
+struct _List_node_base {
+  _List_node_base* _M_next;
+  _List_node_base* _M_prev;
+};
+
+// node struct with data value.
+template <typename _Tp>
+struct _List_node : public _List_node_base {
+  _Tp _M_data;
+};
+
+// list iterator
+template <typename _Tp>
+struct _List_iterator {
+  typedef _List_iterator<_Tp> _Self;
+  typedef _List_node<_Tp> _Node;
+
+  typedef _Tp value_type;
+  typedef _Tp* pointer;
+  typedef _Tp& reference;
+
+  _List_node_base* _M_node;
+
+  // two construct function.
+  _List_iterator() : _M_node() {}
+  explicit _List_iterator(_List_node_base* __x) : _M_node(__x) {}
+
+  // return iterator self.
+  _Self _M_const_cast() const { return *this; }
+
+  // operator functions.
+  reference operator*() const { return static_cast<_Node>(_M_node)->_M_data; }
+  // pointer operator->() const { return }
+  _Self& operator++() {
+    _M_node = _M_node->_M_next;
+    return *this;
+  }
+  _Self operator++(int) {
+    _Self __tmp = *this;
+    _M_node = _M_node->_M_next;
+    return __tmp;
+  }
+  _Self& operator--() {
+    _M_node = _M_node->_M_prev;
+    return *this;
+  }
+  _Self operator--(int) {
+    _Self __tmp = *this;
+    _M_node = _M_node->_M_prev;
+    return __tmp;
+  }
+  bool operator==(const _Self& __x) const { return _M_node == __x._M_node; }
+  bool operator!=(const _Self& __x) const { return _M_node != __x._M_node; }
+};
+#endif
+
+/**
+ * 定义容器空间配置器
+ **/
+// template<typename T>
```

Comment:
Unnecessary code should be deleted

Reply:
Thanks for the suggestion, I've removed the unnecessary code